test/coverage/communication/NotificationSinkTest.cpp

```diff
@@ -126,7 +126,7 @@ TEST_F(NotificationSinkTest, NotificationSinkSuccessWithoutSourceFilter) {
 	EXPECT_TRUE(MsgDiff::Equals(testTopicUUri_, transport->sink_filter_));
 
 	const size_t max_count = 100;
-	for (auto i = 0; i < max_count; i++) {
+	for (size_t i = 0; i < max_count; i++) {
```

Comment:
Minor: an alternate way of writing this
```suggestion
	for (auto remaining = max_count; remaining > 0; --remaining) {
```
This has two benefits:

1. Replaces the `i` index name with a clear name (note: it could also be `remaining_notifications` for greater clarity)
2. `remaining` will always be the same type as `max_count`

Reply:
This is the warning it throws if I use auto

/home/kzpngr/Work/SDV2/opensrc/rc-forks/cpp/up-cpp/test/coverage/communication/NotificationSinkTest.cpp:129:44: error: comparison of integer expressions of different signedness: ‘int’ and ‘const size_t’ {aka ‘const long unsigned int’} [-Werror=sign-compare]
  129 |         for (auto remaining = 0; remaining < max_count; remaining++) {
      |                                  ~~~~~~~~~~^~~~~~~~~~~
cc1plus: all warnings being treated as errors

I think it should be ok to use size_t

Reply:
The warning is correct: you can't use `auto i = 0`. The suggestion was to use `auto i = max_count` which should be fine since `max_count` has an explicit type.